### PR TITLE
`ShallowWaterModel` API documentation + Add all models in Model setup/Overview

### LINKED ISSUE #1708: Model setup/Overview is biased towards `IncompressibleModel`
[Model setup/Overview](https://clima.github.io/OceananigansDocumentation/stable/model_setup/overview/) in Docs only talks about `IncompressibleModel`.

### LINKED ISSUE #1709: Model `struct` definitions should be `mutable`?
Should this
https://github.com/CliMA/Oceananigans.jl/blob/6e1842a20e79c85865e294f1a89032462b4d6dd8/src/Models/ShallowWaterModels/shallow_water_model.jl#L41
be a `mutable struct`, like, e.g.
https://github.com/CliMA/Oceananigans.jl/blob/6e1842a20e79c85865e294f1a89032462b4d6dd8/src/Models/IncompressibleModels/incompressible_model.jl#L22
and
https://github.com/CliMA/Oceananigans.jl/blob/6e1842a20e79c85865e294f1a89032462b4d6dd8/src/Models/HydrostaticFreeSurfaceModels/hydrostatic_free_surface_model.jl#L29
?

### PR DESCRIPTION
**Pull request by navidcy**
This PR adds docstring to `ShallowWaterModel` and also includes all models in Model setup/Overview.

Closes #1708 and closes #1709.

### CODE REVIEW

File: src/Models/ShallowWaterModels/shallow_water_model.jl [hunk 1]
```diff
@@ -66,7 +102,6 @@ function ShallowWaterModel(;
                  forcing::NamedTuple = NamedTuple(),
                              closure = nothing,
                           bathymetry = nothing,
-                            solution = nothing,
```
Comment (navidcy) [inline src/Models/ShallowWaterModels/shallow_water_model.jl:69]: removing this from the constructor seems to not have an effect on the shallow water tests -- so I guess we didn't need it there, right @francispoulin?

**Comment by navidcy** [comment]
any ideas why I get this `undefined binding ShallowWaterModel` error in the [docs](https://buildkite.com/clima/oceananigans/builds/2739#d490498c-b553-4e51-85ae-9d9211617c21/19-315)

File: src/Models/ShallowWaterModels/shallow_water_model.jl (hunk 1 above)
Comment (francispoulin) [reply src/Models/ShallowWaterModels/shallow_water_model.jl:69]: I guess not.  Just to be clear, this line serves the same purpose as [line 36](https://github.com/CliMA/Oceananigans.jl/blob/4411de0ca834351f1c7075a2bee973ecbf64a3f9/src/Models/IncompressibleModels/incompressible_model.jl#L36) of `IncompressibleModel` which specifies the velocities.  I thought we needed this to have the solution as part of the constructor but maybe I'm wrong. 

Can someone explain to me why we might need this in `IncompressibleModel`?

I looked at the test and see that[ line 137 ](https://github.com/CliMA/Oceananigans.jl/blob/4411de0ca834351f1c7075a2bee973ecbf64a3f9/test/test_shallow_water_models.jl#L137) does set the mass fluxes so if that doesn't have a problem then I guess we're fine.

**Comment by francispoulin** [comment]
I can't say I have any idea why this error occurs.  Do we do something special in this doc that we don't do in others?

**Review by francispoulin — APPROVED** [review]
Besides my one question everything here looks great!  Thanks @navidcy for putting this together.

It also reminds me that bathymetry is at the moment just a place holder.  I will create an issue that will actually do something and add a test.

File: src/Models/ShallowWaterModels/shallow_water_model.jl (hunk 1 above)
Comment (glwagner) [reply src/Models/ShallowWaterModels/shallow_water_model.jl:69]: There could be cases where one wants to build `VelocityFields` or tracer fields before constructing the model. In that case users can pass the velocity fields in directly. `velocitices = nothing` specifies the constructor to allocate memory and build the `VelocityFields` internally.

Nothing is "needed", it's just a question of what features are supported. The kwargs are all about supporting features and model designers should choose them with intent.

**Comment by navidcy** [comment]
> I can't say I have any idea why this error occurs. Do we do something special in this doc that we don't do in others?

Got it. It was not exported in the main Oceananigans.jl file. Fixed via 52a394a.

**Pull request merged**